libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/NativeLoginManager.kt

```diff
@@ -270,108 +266,194 @@ internal class NativeLoginManager(
+    ): NativeLoginResult {
+
+        // Validate parameters.
+        val trimmedUsername = username.trim()
+        trimmedUsername.mapInvalidUsernameToResult()?.let { return it }
+        val trimmedNewPassword = newPassword.trim()
+        trimmedNewPassword.mapInvalidPasswordToResult()?.let { return it }
+        val trimmedOtp = otp.trim()
+
+        // Generate the complete password reset request body.
+        val completePasswordResetRequestBodyString = runCatching {
+            CompletePasswordResetRequestBody(
+                username = trimmedUsername,
+                otp = trimmedOtp,
+                newPassword = newPassword
+            ).toJson()
+        }.onFailure { e ->
+            SalesforceSDKLogger.e(TAG, "Cannot JSON encode complete password reset request body due to an encoding error with message '${e.message}'.", e)
+            return UnknownError
+        }.getOrNull()
+
+        // Create the complete password reset request.
+        val completePasswordResetRequest = RestRequest(
+            POST,
+            LOGIN,
+            "$loginUrl$OAUTH_ENDPOINT_HEADLESS_FORGOT_PASSWORD",
+            completePasswordResetRequestBodyString,
+            null
+        )
+
+        // Submit the complete password reset request and fetch the response.
+        val completePasswordResetResponse = suspendedRestCall(completePasswordResetRequest) ?: return UnknownError
+
+        // React to the complete password reset response.
+        return when (completePasswordResetResponse.isSuccess) {
+            true -> {
+                completePasswordResetResponse.consumeQuietly()
+                Success
             }
 
-            else -> googleCloudProjectId
+            else -> {
+                SalesforceSDKLogger.e(
+                    TAG,
+                    "Cannot complete password reset as the Salesforce Identity API returned an unsuccessful HTTP status. '${completePasswordResetResponse.asString()}'"
+                )
+                UnknownError
+            }
         }
+    }
 
-        /*
-         * Create the OTP request body with the provided parameters. Note: The
-         * `emailtemplate` parameter isn't supported here, but could be added in
-         * the future.
-         */
-        // Determine the reCAPTCHA parameter for non-enterprise reCAPTCHA
-        val reCaptchaParameter = when {
-            isReCaptchaEnterprise -> null
-            else -> reCaptchaToken
-        }
-        // Determine the reCAPTCHA "event" parameter for enterprise reCAPTCHA
-        val reCaptchaEventParameter = when {
-            isReCaptchaEnterprise -> OtpRequestBodyReCaptchaEvent(
-                token = reCaptchaToken,
-                siteKey = reCaptchaSiteKeyId,
-                projectId = googleCloudProjectId
-            )
+    // endregion
+    // region Salesforce Identity API Headless, Password-Less Login Via One-Time-Passcode
+
+    override suspend fun startPasswordLessAuthorization(
```

Comment:
Here's a question: In 12.0 I rather naively named this `submitOtpRequest`.  Now I realize that's very ambiguous in the larger scope of multiple Salesforce Identity flows that will request and use an OTP.  Also, it doesn't help the app developer know which comes first in the start and completion of this two-step flow.

Without breaking the API in 12.1, I deprecated the former method and made it a proxy to call the newer method.  The app code is _much_ more intuitive afterwards.  Any objections to that in general or in a point release?  It might be nice to do this before adoption takes off.

I could roll this back if we feel it's a blocker.

Reply:
As I said in my other comment I am not a huge fan or renaming immediately.  If we are going to rename I just think we should consider multiple options and potentially have a more cohesive naming scheme for all the interface APIs.  We should also consider that we could wait an rename these in the future if/when this functionality expands beyond communities -- which will require us to at least update if not recreate the API.

Reply:
I simply reverted this in the interest of expediency 🏁